### DISCUSSION TITLE
jvm/whereis-the-object

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#217] [Announcements]
# jvm/whereis-the-object

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，Java 创建的对象到底放在哪？

https://tobebetterjavaer.com/jvm/whereis-the-object.html

**Comment by hxsdxwch0520** [comment]
动态年龄判断，应该不是相同年龄的所有对象的总和大于一半吧。下来的解释是否正确啊？？？？
我看了其他的地方解释为：
年龄1+年龄2+年龄3+年龄N的对象加起来的空间，大于survivor区域的一半，就会让年龄N和年龄N以上的对象进入老年代。动态年龄判断应该是这样子的。说的通俗一点：就是年龄从小到大对象的占据空间的累加和，而不是某一个特定年龄对象占据的空间。

**Reply by fantasy-zrq** [reply]
确实不对，举个例子，假设survival0区的大小是10M，里面对象的年龄1岁的占用2M，2岁的占用2M，3岁的占用2M，那么此时1，2，3岁的对象已经占据6M>5M(survival0区内存的一半)。所以在剩余的4岁，5岁.......等对象都会直接进入老年代。

**Comment by zxysdtc** [comment]
设置年龄的指令好像错了

**Comment by Raceylee** [comment]
“比如 Survivor 是 100M，Hello1 和 Hello2 都是 3 岁，且总和超过了 50M，Hello3 是 4 岁，这个时候，这三个对象都将到老年代。”
确实在不少地方看到本文这种说法，但是这些三岁的对象总和超过一半，难道不是第一次进入 survivor 区时候就已经超过一半了吗，同一个年龄的对象只会越来越少吧
